### ISSUE TITLE
Add ICMPListener

### ISSUE THREAD

**Issue by xmh0511** [automesh-network/netstack-smoltcp#8]
Seems to lack the ICMPListener, which is used to process ICMP packets over `netstack-smoltcp`.

**Comment by cavivie** [comment]
Yep, PRs are welcome. I think TCP/UDP/ICMP Listeners should be features so that people can enable them optionally.
